internal/infrastructure/ceph/rbd.go

```diff
@@ -66,6 +70,75 @@ func (r *RBDRepository) ListSnapshots(poolName, imageName string) ([]*model.RBDS
 }
 
 func (r *RBDRepository) ExportDiff(input *model.ExportDiffInput) error {
+	stream, err := r.startExportDiffStream(input)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if stream != nil {
+			_ = stream.Close()
+		}
+	}()
+
+	outputFile, err := os.OpenFile(input.OutputFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to create output file %s: %w", input.OutputFile, err)
+	}
+	defer func() { _ = outputFile.Close() }()
+
+	checksumPath := csumio.GetChecksumFilePath(input.OutputFile)
+	checksumFile, err := os.OpenFile(checksumPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to create checksum file %s: %w", checksumPath, err)
+	}
+	defer func() { _ = checksumFile.Close() }()
+
+	chunkSize := input.DiffChecksumChunkSize
+	if chunkSize == 0 {
+		chunkSize = model.DefaultDiffChecksumChunkSize
+	}
+
+	writer, err := csumio.NewWriter(outputFile, checksumFile, int(chunkSize))
+	if err != nil {
+		return fmt.Errorf("failed to create checksum writer: %w", err)
+	}
+	if _, err := io.Copy(writer, stream); err != nil {
+		return fmt.Errorf("failed to write diff to %s: %w", input.OutputFile, err)
+	}
+	if err := writer.Close(); err != nil {
+		return fmt.Errorf("failed to finalize diff checksum: %w", err)
+	}
+	if err := stream.Close(); err != nil {
+		return fmt.Errorf("failed to finalize export diff: %w", err)
+	}
+
+	return nil
+}
+
+type commandReadCloser struct {
+	io.ReadCloser
+	cmd    *exec.Cmd
+	op     string
+	stderr *bytes.Buffer
+}
+
+func (c *commandReadCloser) Close() error {
+	if c.ReadCloser != nil {
+		_ = c.ReadCloser.Close()
+		c.ReadCloser = nil
+	}
```

Comment:
We don't have to call `c.ReadCloser.Close()` because `cmd.Wait` closes it. Instead we need to drain all the data from the reader before calling `cmd.Wait`.
cf. https://pkg.go.dev/os/exec#Cmd.StdoutPipe